drivers/src/hal_sfc.c

```diff
@@ -1266,7 +1289,34 @@ static int spiflash_set_qpi(struct spi_flash *flash, u32_t op)
 	}
 }
 
-__ramfunc static void spiflash_enter_xip(void)
+static int spiflash_read(struct spi_flash *flash, u32_t offset,
+		u32_t *buf, u32_t len, READ_CMD_TYPE_E type)
+{
+
+	int ret = TRUE;
+
+#ifndef CONFIG_W25_READONLY
+
+	u8_t dump_byte = 1;
+	u32_t read_char = 0;
+
+	/* modify for xip-sfc */
+
+	ret = spiflash_read_data_xip(flash, offset, (u32_t *) &read_char,
+				     dump_byte, type);
+
+
+	memcpy((char *)buf, (char *)read_char, len);
+
+#else
+	LOG_INF("defined CONFIG_W25_READONLY\n");
+	memcpy((char *)buf, (char *)(offset), len);
+#endif
+
+	return ret;
+}
+
+__ramfunc static void spiflash_exit_xip(void)
 {
 	/* pin mux and drive configuration */
 	sci_write32(0x40840038, 0x4100);
```

Comment:
Can you define all registers and value with macro, please look at drivers/src/hal_pinmux.c for reference.